iothub/device/src/Common/Exceptions/DeviceAlreadyExistsException.cs

```diff
@@ -1,56 +0,0 @@
-// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
-using System;
-using System.Runtime.Serialization;
-using Microsoft.Azure.Devices.Client.Extensions;
-
-namespace Microsoft.Azure.Devices.Client.Exceptions
-{
-    /// <summary>
-    /// The exception that is thrown when an attempt to create a device fails because it already exists.
-    /// </summary>
-    [Serializable]
-    public sealed class DeviceAlreadyExistsException : IotHubException
```

Comment:
This type was not referenced anywhere in the SDK -> no exception returned by service was mapped to this type.
@prmathur-microsoft , @sharmasejal This is a public type that we are removing.